config/config.exs

```diff
@@ -22,19 +22,22 @@ use Mix.Config
 #
 
 # Use rails credentials:edit to set the AWS secrets (as aws:access_key_id|secret_access_key)
-config :active_storage, :storage,
-  amazon: %{
-    service: "S3",
-    region: System.get_env("AWS_S3_REGION"),
-    access_key_id: System.get_env("AWS_ACCESS_KEY_ID"),
-    secret_access_key: System.get_env("AWS_SECRET_ACCESS_KEY"),
-    bucket: System.get_env("AWS_BUCKET")
+config :active_storage, :storage, %{
+  s3: %{
+    bucket: "active-storage-test",
+    access_key_id: "root",
+    secret_access_key: "active_storage_test",
+    scheme: "http://",
+    host: "localhost",
+    port: 9000,
+    force_path_style: true
```

Comment:
I would keep the format of activestorage configuration; I think we could use the endpoint config attribute here to point to the Minio service. example: https://kevinjalbert.com/rails-activestorage-configuration-for-minio/
let me know your thoughts on this.

Reply:
Ah, sure, that makes sense.  That has keys for aws in general, though there are some S3-specific in here (`bucket` and I think maybe `force_path_style`)  Though I guess that's how it is in the official activestorage docs: https://edgeguides.rubyonrails.org/active_storage_overview.html#setup

Actually, I guess that those keys (`local` / `test` / `amazon`) are just keys that you can refer to to choose the configuration.  In this PR there's actually references to the service to override the default so that we can have different services for different attachments: https://github.com/chaskiq/active_storage_ex/blob/5e8a4e32ec01120994fb2baffd36c3078a40c10d/test/app/app/models/record.rb#L6

Reply:
Ok, have another look.  I think I've found out a somewhat better way to deal with configurations.  Called them `sources` to distinguish from `services`.  Trying to copy what `activestorage` does (but in a functional way).

Added a `purge_attachment` function to delete files (from the DB and source).  Still need to test it and then put it inside a transaction.

Also, trying out making `ActiveStorage.Service` a behavior.

Reply:
awesome, looks good to me!